### PR TITLE
MRG: Add SSP projectors to Report

### PR DESCRIPTION
**Pull request by hoechenberger**
This is still WIP and depends on #7990. So will need to be rebased after #7990 has been merged.

### CODE REVIEW

**Comment by agramfort** [comment]
can you share a rendered report to see how it looks?

**Comment by hoechenberger** [comment]
> can you share a rendered report to see how it looks?

Yes, once it renders :D

**Comment by hoechenberger** [comment]
Built doc: https://21081-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/misc/plot_report.html#sphx-glr-auto-tutorials-misc-plot-report-py

See the 2nd Report in there for an example with SSP projectors.

**Comment by hoechenberger** [comment]
built doc:
https://21097-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/misc/plot_report.html#sphx-glr-auto-tutorials-misc-plot-report-py

**Comment by hoechenberger** [comment]
@larsoner @GuillaumeFavelier Any idea why the notebook test is failing? Seems to be unrelated to this PR?

**Comment by agramfort** [comment]
@hoechenberger projs are often stored in separate files like a cov.fif you have proj.fif. Cf sample data and study template that have projs for ecg and eog. Here you plot the projs in the data. Please add the other option with separate files.

**Comment by hoechenberger** [comment]
Doc artifact:
https://21141-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/misc/plot_report.html#sphx-glr-auto-tutorials-misc-plot-report-py

**Comment by hoechenberger** [comment]
re Travis error see https://github.com/mne-tools/mne-python/pull/8000#issuecomment-657131358

File: mne/tests/test_report.py [hunk 1]
```diff
@@ -83,7 +86,8 @@ def test_render_report(renderer, tmpdir):
     # Also, make sure crop range is wide enough to avoid rendering bug
     epochs.average().crop(0.1, 0.2).save(evoked_fname)
 
-    report = Report(info_fname=raw_fname_new, subjects_dir=subjects_dir)
+    report = Report(info_fname=raw_fname_new, subjects_dir=subjects_dir,
+                    projs=True)
```
Comment (agramfort) [inline mne/tests/test_report.py:90]: you should check that the proj appears in the rendered html. See how it's done elsewhere. I would just check that a certain string is present.

Comment (hoechenberger) [reply mne/tests/test_report.py:91]: Done in d9cc5cd

**Comment by hoechenberger** [comment]
Travis errors seem unrelated. We need to fix this flake8 thing for Python 3.6…

**Comment by hoechenberger** [comment]
As far as I am concerned, this is good to merge.

File: mne/report.py [hunk 2]
```diff
@@ -1615,10 +1647,15 @@ def __exit__(self, type, value, traceback):
...
-            caption = f'{prefix}: {fname} {suffix}'
-        else:
-            caption = f'{prefix}: {fname[len(data_path)+1:]} {suffix}'
+        if data_path is not None:
+            # Ensure we drop leading path separators from the
+            # filename before using it as a caption. We don't know whether
+            # `data_path` has a trailing separator or not; so create a Path
+            # just to be safe (trailing separators will be removed during the
+            # roundtrip).
+            fname = fname[len(str(Path(data_path))) + 1:]
```
Comment (larsoner) [inline mne/report.py:1656]: `os.path.relpath` instead?

Comment (hoechenberger) [reply mne/report.py:1656]: No it would have a different behavior than what I'm trying to achieve here… I know it's not pretty, but it does to the job.

Comment (larsoner) [reply mne/report.py:1656]: When would it be different?

Comment (hoechenberger) [reply mne/report.py]: Never mind, `op.relpath` makes total sense here. Thanks!

**Comment by hoechenberger** [comment]
I have no idea why the phsionet dataset test is failing on Travis, all dataset tests pass without any issues locally for me.

**Comment by larsoner** [comment]
Thanks @hoechenberger

**Pull request merged**